apps/cli/src/lib/ProjectTools.ts

```diff
@@ -0,0 +1,54 @@
+import dotenv from "dotenv";
+import * as fs from "node:fs/promises";
+import { DisployConfig, readConfig } from "./disployConf";
+let config: DisployConfig | undefined;
+
+async function resolveProject({ cwd }: { cwd: string }) {
+  if (!config) {
+    const pkg = await readConfig(cwd);
+    config = pkg;
+  }
+
+  return config;
+}
+
+async function findClosestEnv() {
+  let cwd = process.cwd();
+
+  while (cwd !== "/") {
+    const envPath = `${cwd}/.env`;
+    try {
+      await fs.access(envPath);
+      return envPath;
+    } catch (e) {
+      cwd = cwd.substring(0, cwd.lastIndexOf("/"));
+    }
+  }
+
+  return undefined;
+}
```

Comment:
This can be modeled more elegantly as a recursive function. You'll need to import `fs from "node:fs/promises"`
```suggestion
async function findClosestEnv(curPath=process.cwd()) {
	if (curPath === "/") {
		return undefined;
	}
	
	// Check if the env is on this level
	const envPath = path.join(curPath, ".env");
	const stat = await fs.stat(envPath);
	
	if (stat.isFile()) {
		return curPath;
	}
	
	// Otherwise move a dir up
	const nextDir = path.join(curPath, "..");
	return findClosestEnv(nextDir);
}
```